thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/detector/email/filter/AlertFilterFactory.java

```diff
@@ -0,0 +1,83 @@
+package com.linkedin.thirdeye.detector.email.filter;
+
+import com.linkedin.thirdeye.datalayer.dto.AnomalyFunctionDTO;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Collections;;
+import java.util.Map;
+import java.util.Properties;
+import org.apache.commons.io.IOUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+
+public class AlertFilterFactory {
+  private static Logger LOGGER = LoggerFactory.getLogger(AlertFilterFactory.class);
+  public static final String FILTER_TYPE_KEY = "type";
+  private final Properties props;
+
+
+  public AlertFilterFactory(String AlertFilterConfigPath) {
+    props = new Properties();
+    try {
+      InputStream input = new FileInputStream(AlertFilterConfigPath);
+      loadPropertiesFromInputStream(input);
+    } catch (FileNotFoundException e) {
+      LOGGER.error("Alert Filter Property File {} not found", AlertFilterConfigPath, e);
+    }
+  }
+
+  public AlertFilterFactory(InputStream input) {
+    props = new Properties();
+    loadPropertiesFromInputStream(input);
+  }
+
+  private void loadPropertiesFromInputStream(InputStream input) {
+    try {
+      props.load(input);
+    } catch (IOException e) {
+      LOGGER.error("Error loading the alert filters from config", e);
+    } finally {
+      IOUtils.closeQuietly(input);
+    }
+
+    LOGGER.info("Found {} entries in alert filter configuration file {}", props.size());
+    for (Map.Entry<Object, Object> entry : props.entrySet()) {
+      LOGGER.info("{}: {}", entry.getKey(), entry.getValue());
+    }
+  }
+
+
+  /**
+   * Initiates an alert filter for the given anomaly function.
+   *
+   * @param anomalyFunctionSpec the anomaly function that contains the alert filter spec.
+   *
+   * @return the alert filter specified by the alert filter spec or a dummy filter if the function
+   * does not have an alert filter spec or this method fails to initiates an alert filter from the
+   * spec.
+   */
+  public AlertFilter getAlertFilter(AnomalyFunctionDTO anomalyFunctionSpec) {
```

Comment:
Should this method's signature be similar to AlertFilterAutoTuneFactory's fromSpec()? It takes the spec to the AlertFilter instead of anomaly function spec. You could use a helper class to extract the alertFilter spec from anomaly function spec and then invokes this method using alertFilter spec as the input.

You could rename this method to fromSpec() to be consistent with the method from AnomalyFunctionFactory and AlertFilterAutoTuneFactory.